File: Models/Fine.cs
namespace LibraryManagementSystem.Models;

public class Fine
{
    public int Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public int BorrowRecordId { get; set; }
    public decimal Amount { get; set; }
    public bool IsPaid { get; set; } = false;
    public DateTime IssuedAt { get; set; } = DateTime.UtcNow;
    public DateTime? PaidAt { get; set; }
    public User User { get; set; } = null!;
    public BorrowRecord BorrowRecord { get; set; } = null!;
}

File: app/models/fine.py
from app import db
from datetime import datetime

class Fine(db.Model):
    __tablename__ = 'fines'
    id = db.Column(db.Integer, primary_key=True)
    borrow_record_id = db.Column(db.Integer, db.ForeignKey('borrow_records.id'), nullable=False)
    amount = db.Column(db.Float, nullable=False)
    is_paid = db.Column(db.Boolean, default=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)

    def to_dict(self):
        return {
            'id': self.id,
            'borrow_record_id': self.borrow_record_id,
            'amount': self.amount,
            'is_paid': self.is_paid,
            'created_at': self.created_at.isoformat()
        }
